src/main/java/org/csanchez/jenkins/plugins/kubernetes/volumes/SecretVolume.java

```diff
@@ -33,21 +33,32 @@
 import io.fabric8.kubernetes.api.model.VolumeBuilder;
 
 public class SecretVolume extends PodVolume {
+    private static final String DEFAULT_MODE = "420";
 
     private String mountPath;
     private String secretName;
+    private String defaultMode;
 
     @DataBoundConstructor
-    public SecretVolume(String mountPath, String secretName) {
+    public SecretVolume(String mountPath, String secretName, String defaultMode) {
         this.mountPath = mountPath;
         this.secretName = secretName;
+        if (defaultMode == null) {
+            this.defaultMode = DEFAULT_MODE;
+        } else {
+            this.defaultMode = defaultMode;
+        }
+    }
+
+    public SecretVolume(String mountPath, String secretName) {
+        this(mountPath, secretName, DEFAULT_MODE);
     }
 
     @Override
     public Volume buildVolume(String volumeName) {
         return new VolumeBuilder()
                 .withName(volumeName)
-                .withNewSecret().withSecretName(getSecretName()).endSecret()
+                .withNewSecret().withSecretName(getSecretName()).withDefaultMode(Integer.parseInt(defaultMode)).endSecret()
```

Comment:
This could be just Github's rendering, but this line looks a tad bit long.

Reply:
Adjusted the spacing a bit 👍Everything should fit now.